[Bug] Build and Test Scheduled On 3.3 might be always failed since 'feat: remove useless dependency(#15125)'

### Pre-check

- [x] I am sure that all the content I provide is in English.


### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/dubbo/issues?q=is%3Aissue) and found no similar issues.


### Apache Dubbo Component

Java SDK (apache/dubbo)

### Dubbo Version

dubbo 3.x


### Steps to reproduce this issue

see ```Build and Test Scheduled On 3.3``` at https://github.com/apache/dubbo/actions

Almost all ```Unit Test On windows-latest``` are failed since feat: remove useless dependency(#15125), i doubt dubbo-test-check might not be removed from some test modules or ReplierDispatcherTest might have some issues about threadpool size.
```
Error:  Errors: 
Error:    ReplierDispatcherTest.testMultiThread:114 Execution org.opentest4j.AssertionFailedError
```
@Stellar1999  @CrazyHZM 

### What you expected to happen

Build and Test Scheduled On 3.3 should be successful.

### Anything else

_No response_

### Are you willing to submit a pull request to fix on your own?

- [x] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

extracted a error message snippet from https://github.com/apache/dubbo/actions/runs/13425007540/job/37506574404
```
Error:  org.apache.dubbo.remoting.transport.netty4.ReplierDispatcherTest.testMultiThread -- Time elapsed: 11.06 s <<< ERROR!
java.util.concurrent.ExecutionException: org.opentest4j.AssertionFailedError
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at org.apache.dubbo.remoting.transport.netty4.ReplierDispatcherTest.testMultiThread(ReplierDispatcherTest.java:114)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: org.opentest4j.AssertionFailedError
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:46)
	at org.junit.jupiter.api.Assertions.fail(Assertions.java:161)
	at org.apache.dubbo.remoting.transport.netty4.ReplierDispatcherTest.lambda$testMultiThread$1(ReplierDispatcherTest.java:110)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.util.concurrent.ExecutionException: org.apache.dubbo.remoting.RemotingException: Server side(127.0.0.1,31708) thread pool is exhausted, detail msg:Thread pool is EXHAUSTED! Thread Name: DubboServerHandler-127.0.0.1:8080, Pool Size: 200 (active: 200, core: 200, max: 200, largest: 200), Task: 499 (completed: 299), Executor status:(isShutdown:false, isTerminated:false, isTerminating:false), in empty://127.0.0.1:8080!
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at org.apache.dubbo.remoting.transport.netty4.ReplierDispatcherTest.clientExchangeInfo(ReplierDispatcherTest.java:136)
	at org.apache.dubbo.remoting.transport.netty4.ReplierDispatcherTest.lambda$testMultiThread$1(ReplierDispatcherTest.java:108)
	... 5 more
Caused by: org.apache.dubbo.remoting.RemotingException: Server side(127.0.0.1,31708) thread pool is exhausted, detail msg:Thread pool is EXHAUSTED! Thread Name: DubboServerHandler-127.0.0.1:8080, Pool Size: 200 (active: 200, core: 200, max: 200, largest: 200), Task: 499 (completed: 299), Executor status:(isShutdown:false, isTerminated:false, isTerminating:false), in empty://127.0.0.1:8080!
	at org.apache.dubbo.remoting.exchange.support.DefaultFuture.doReceived(DefaultFuture.java:265)
	at org.apache.dubbo.remoting.exchange.support.DefaultFuture.received(DefaultFuture.java:208)
	at org.apache.dubbo.remoting.exchange.support.DefaultFuture.received(DefaultFuture.java:196)
	at org.apache.dubbo.remoting.exchange.support.header.HeaderExchangeHandler.handleResponse(HeaderExchangeHandler.java:65)
	at org.apache.dubbo.remoting.exchange.support.header.HeaderExchangeHandler.received(HeaderExchangeHandler.java:211)
	at org.apache.dubbo.remoting.transport.DecodeHandler.received(DecodeHandler.java:52)
	at org.apache.dubbo.remoting.transport.dispatcher.ChannelEventRunnable.run(ChannelEventRunnable.java:64)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at org.apache.dubbo.common.threadlocal.InternalRunnable.run(InternalRunnable.java:39)
	... 1 more
```

--- a/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/RpcMessageHandler.java
+++ b/dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/RpcMessageHandler.java
@@ -18,26 +18,32 @@ package org.apache.dubbo.remoting.transport.netty4;
 
 import org.apache.dubbo.common.bytecode.NoSuchMethodException;
 import org.apache.dubbo.common.bytecode.Wrapper;
+import org.apache.dubbo.common.utils.ConcurrentHashMapUtils;
 import org.apache.dubbo.remoting.RemotingException;
 import org.apache.dubbo.remoting.exchange.ExchangeChannel;
 import org.apache.dubbo.remoting.exchange.support.Replier;
 
 import java.lang.reflect.InvocationTargetException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 
 /**
  * RpcMessageHandler.
  */
 public class RpcMessageHandler implements Replier<RpcMessage> {
+    private static final ConcurrentMap<String, Object> IMPLEMENT_MAP = new ConcurrentHashMap<>();
     private static final ServiceProvider DEFAULT_PROVIDER = new ServiceProvider() {
         public Object getImplementation(String service) {
             String impl = service + "Impl";
-            try {
-                Class<?> cl = Thread.currentThread().getContextClassLoader().loadClass(impl);
-                return cl.getDeclaredConstructor().newInstance();
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
-            return null;
+            return ConcurrentHashMapUtils.computeIfAbsent(IMPLEMENT_MAP, impl, (s) -> {
+                try {
+                    Class<?> cl = Thread.currentThread().getContextClassLoader().loadClass(s);
+                    return cl.getDeclaredConstructor().newInstance();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+                return null;
+            });
         }
     };
     private ServiceProvider mProvider;

